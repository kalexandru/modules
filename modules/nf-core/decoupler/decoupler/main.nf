process DECOUPLER {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/7d/7d304d1b25aa80ce8c44c8d34ea45d6b4f6f50697a4effcf1b9be4a54db19928/data' :
        'community.wave.seqera.io/library/decoupler-py_matplotlib_pandas_pip_pruned:0d4681dad9987ec5' }"

    input:
    tuple val(meta), path(mat)
    path(net)

    output:
    tuple val(meta), path("*estimate__decoupler.tsv"), emit: dc_estimate
    tuple val(meta), path("*pvals__decoupler.tsv"), emit: dc_pvals
    path("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'decoupler.py'

    stub:
    """
    touch mlm_estimate__decoupler.tsv
    touch mlm_pvals__decoupler.tsv
    touch versions.yml
    """
}
