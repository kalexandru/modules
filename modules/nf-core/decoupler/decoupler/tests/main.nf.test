nextflow_process {

    name "Test Process DECOUPLER"
    script "../main.nf"
    config "./nextflow.config"
    process "DECOUPLER"

    tag "modules"
    tag "modules_nfcore"
    tag "decoupler"
    tag "decoupler/decoupler"

    test("decoupler_test") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.test_data['generic']['tsv']['expression'], checkIfExists: true)
                ]
                input[1] = [
                    file(params.test_data['generic']['tsv']['network'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("progeny") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'progeny' ],
                    file("https://github.com/atrigila/test-datasets/raw/progeny/data/delete_me/decoupler/progeny/pbmc3k_1000rows.tsv", checkIfExists: true)
                ]
                input[1] = [
                    file("https://github.com/atrigila/test-datasets/raw/progeny/data/delete_me/decoupler/progeny/progeny_data.tsv", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("decoupler stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.test_data['generic']['tsv']['expression'], checkIfExists: true)
                ]
                input[1] = [
                    file(params.test_data['generic']['tsv']['network'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
