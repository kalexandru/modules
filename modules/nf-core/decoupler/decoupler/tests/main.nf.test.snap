{
    "progeny": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "progeny"
                        },
                        [
                            "consensus_estimate__decoupler.tsv:md5,44953b6d630644f37b18db249625d242",
                            "mlm_estimate__decoupler.tsv:md5,9b7dac003edb0365b4ebcec7f34bb596",
                            "ulm_estimate__decoupler.tsv:md5,184529b96914c3f62518fdedc9b955d3",
                            "wsum_estimate__decoupler.tsv:md5,c3ca1702a2dc5168d42074196167cbe1"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "progeny"
                        },
                        [
                            "consensus_pvals__decoupler.tsv:md5,0dbe30434b90e8c3653e123509d4ead8",
                            "mlm_pvals__decoupler.tsv:md5,24183d3de950603a429bfb7cfb049e45",
                            "ulm_pvals__decoupler.tsv:md5,8937297c31e1739a86029c602a58b426",
                            "wsum_pvals__decoupler.tsv:md5,11c96cc3e9e128b0b1ba000c66ddfa7b"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,0b46461966f127c5bacbf7aa4c447097"
                ],
                "dc_estimate": [
                    [
                        {
                            "id": "progeny"
                        },
                        [
                            "consensus_estimate__decoupler.tsv:md5,44953b6d630644f37b18db249625d242",
                            "mlm_estimate__decoupler.tsv:md5,9b7dac003edb0365b4ebcec7f34bb596",
                            "ulm_estimate__decoupler.tsv:md5,184529b96914c3f62518fdedc9b955d3",
                            "wsum_estimate__decoupler.tsv:md5,c3ca1702a2dc5168d42074196167cbe1"
                        ]
                    ]
                ],
                "dc_pvals": [
                    [
                        {
                            "id": "progeny"
                        },
                        [
                            "consensus_pvals__decoupler.tsv:md5,0dbe30434b90e8c3653e123509d4ead8",
                            "mlm_pvals__decoupler.tsv:md5,24183d3de950603a429bfb7cfb049e45",
                            "ulm_pvals__decoupler.tsv:md5,8937297c31e1739a86029c602a58b426",
                            "wsum_pvals__decoupler.tsv:md5,11c96cc3e9e128b0b1ba000c66ddfa7b"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0b46461966f127c5bacbf7aa4c447097"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-16T22:46:06.40580042"
    },
    "decoupler stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "mlm_estimate__decoupler.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "mlm_pvals__decoupler.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "dc_estimate": [
                    [
                        {
                            "id": "test"
                        },
                        "mlm_estimate__decoupler.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "dc_pvals": [
                    [
                        {
                            "id": "test"
                        },
                        "mlm_pvals__decoupler.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-17T13:51:24.782790773"
    },
    "decoupler_test": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "consensus_estimate__decoupler.tsv:md5,f762d3282a81c61b127e97d044e9ddd7",
                            "mlm_estimate__decoupler.tsv:md5,ce7ce3c710fe188035b6706512ffbf54",
                            "ulm_estimate__decoupler.tsv:md5,14773953134037a208fe3c95c893acae",
                            "wsum_estimate__decoupler.tsv:md5,2488716cfe85c1306cb76cf117808d97"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "consensus_pvals__decoupler.tsv:md5,35850081a4308deba35d171a266bedd0",
                            "mlm_pvals__decoupler.tsv:md5,016a15c3dcf688fcdf3ccbd40e06d64f",
                            "ulm_pvals__decoupler.tsv:md5,531a3ed641bf3c1b0b105cfaee5864cd",
                            "wsum_pvals__decoupler.tsv:md5,fac9e18b9495ca8a9ae92a015eed3043"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,0b46461966f127c5bacbf7aa4c447097"
                ],
                "dc_estimate": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "consensus_estimate__decoupler.tsv:md5,f762d3282a81c61b127e97d044e9ddd7",
                            "mlm_estimate__decoupler.tsv:md5,ce7ce3c710fe188035b6706512ffbf54",
                            "ulm_estimate__decoupler.tsv:md5,14773953134037a208fe3c95c893acae",
                            "wsum_estimate__decoupler.tsv:md5,2488716cfe85c1306cb76cf117808d97"
                        ]
                    ]
                ],
                "dc_pvals": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "consensus_pvals__decoupler.tsv:md5,35850081a4308deba35d171a266bedd0",
                            "mlm_pvals__decoupler.tsv:md5,016a15c3dcf688fcdf3ccbd40e06d64f",
                            "ulm_pvals__decoupler.tsv:md5,531a3ed641bf3c1b0b105cfaee5864cd",
                            "wsum_pvals__decoupler.tsv:md5,fac9e18b9495ca8a9ae92a015eed3043"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0b46461966f127c5bacbf7aa4c447097"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-16T22:43:18.532694254"
    }
}